codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -314,11 +327,13 @@ public void onPriorityRead(ChannelHandlerContext ctx, int streamId, int streamDe
             Http2Stream stream = connection.stream(streamId);
 
             try {
-                if (stream == null) {
+                if (stream == null && !connection.streamMayHaveExisted(streamId)) {
                     // PRIORITY frames always identify a stream. This means that if a PRIORITY frame is the
                     // first frame to be received for a stream that we must create the stream.
                     stream = connection.remote().createStream(streamId);
-                } else if (streamCreatedAfterGoAwaySent(stream)) {
+                }
+
+                if (stream == null || streamCreatedAfterGoAwaySent(streamId)) {
```

Comment:
It's more that I was just trying to consolidate the places where we ignore the frame.  I'm fine with your logic ... done.